### PR TITLE
adds tests for the missing members of RTCRtpHeaderExtensionParameters

### LINKED ISSUE #10253: RTCRtpHeaderExtensionParameters missing members are not tested
https://github.com/w3c/webrtc-pc/pull/1816 made two members of [RTCRtpHeaderExtensionParameters](https://w3c.github.io/webrtc-pc/#dom-rtcrtpheaderextensionparameters) required, and this could be tested here:
https://github.com/w3c/web-platform-tests/blob/172fe9d52634f70dd2810c4a859acd1aa6dc365a/webrtc/RTCRtpParameters-headerExtensions.html#L58

(This test passes all the members, 3 tests that each omit one of them would do the trick.)

### PR DESCRIPTION
**Pull request by kritisingh1**
Closes #10253 
@foolip please review and let me know further changes required, if any.
Thanks

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/26775)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/361710503)

**Comment by kritisingh1** [comment]
@jgraham could you please review this and tell me if any further changes are required? Thanks :)

File: webrtc/RTCRtpParameters-headerExtensions.html [hunk 1]
```diff
@@ -63,13 +63,44 @@
...
+
+    param.headerExtensions = [{
+      uri: 'non-existent.example.org',
+      encrypted: false
+    }];
+
+    return promise_rejects(t, 'InvalidModificationError',
+      sender.setParameters(param));
+
+  }, `setParameters() with modified headerExtensions should reject with InvalidModificationError, checks that id is a required field`);
```
Comment (alvestrand) [inline webrtc/RTCRtpParameters-headerExtensions.html]: This test doesn't test anything at all. You can't add new header extensions, and the error code (InvalidModificationError) is the same for all cases.

**Review by alvestrand — CHANGES_REQUESTED** [review]
Sorry, I don't think that this proposal for more tests makes sense.
It's not possible to add RTP extensions that the browser doesn't support - they require code deep in the WebRTC RTP engine, which you can't add through the API.
See http://w3c.github.io/webrtc-pc/#rtcrtpparameters - the header extensions is a read-only parameter.
I don't have any idea how you can verify that these mandatory fields in a read-only parameter are mandatory, but this approach won't work.

**Comment by alvestrand** [comment]
Closing this PR, since a different approach is required (and I don't know what it is).

**Pull request closed without merging**